addons/website/models/ir_http.py

```diff
@@ -61,7 +61,7 @@ def _auth_method_public(cls):
         if not request.session.uid:
             env = api.Environment(request.cr, SUPERUSER_ID, request.context)
             website = env['website'].get_current_website()
-            if website:
+            if website.user_id:
```

Comment:
The original code implies that website can be an empty recordset, if that is the case, accessing its user_id will return another empty recordset and when trying to get the id of that empty recordset it will return False, meaning that the bug is still there, request.uid can still be set to False.

Check for both website and website.user_id